File: doc/api-examples/python/start-unit.py
#!/usr/bin/python3
#
# Copyright Contributors to the Eclipse BlueChi project
#
# SPDX-License-Identifier: MIT-0

import sys

from dasbus.connection import SystemMessageBus

bus = SystemMessageBus()

if len(sys.argv) != 3:
    print(f"Usage: python {sys.argv[0]} node_name unit_name")
    sys.exit(1)

node_name = sys.argv[1]
unit_name = sys.argv[2]

controller = bus.get_proxy("org.eclipse.bluechi", "/org/eclipse/bluechi")
node_path = controller.GetNode(node_name)
node = bus.get_proxy("org.eclipse.bluechi", node_path)

my_job_path = node.StartUnit(unit_name, "replace")

print(f"Started unit '{unit_name}' on node '{node_name}': {my_job_path}")

File: doc/api-examples/cpp/start-unit.cpp
/*
 * Copyright Contributors to the Eclipse BlueChi project
 *
 * SPDX-License-Identifier: MIT-0
 */
#include <iostream>
#include <sdbus-c++/sdbus-c++.h>

int main(int argc, char *argv[]) {
        if (argc != 3) {
                std::cerr << "Usage: " << argv[0] << " node_name unit_name" << std::endl;
                return 1;
        }

        char *nodeName = argv[1];
        char *unitName = argv[2];

        auto connection = sdbus::createSystemBusConnection();
        auto controllerProxy = sdbus::createProxy(
                        *connection,
                        sdbus::ServiceName("org.eclipse.bluechi"),
                        sdbus::ObjectPath("/org/eclipse/bluechi"));

        sdbus::ObjectPath nodePath;
        controllerProxy->callMethod("GetNode")
                        .onInterface("org.eclipse.bluechi.Controller")
                        .withArguments(nodeName)
                        .storeResultsTo(nodePath);

        auto nodeProxy = sdbus::createProxy(
                        *connection, sdbus::ServiceName("org.eclipse.bluechi"), sdbus::ObjectPath(nodePath));

        sdbus::ObjectPath myJobPath;
        nodeProxy->callMethod("StartUnit")
                        .onInterface("org.eclipse.bluechi.Node")
                        .withArguments(unitName, "replace")
                        .storeResultsTo(myJobPath);

        std::cout << "Started unit '" << unitName << "' on node '" << nodeName << "': " << myJobPath
                  << std::endl;

        return 0;
}
